jobs/secureproxy/templates/config/nginx.conf.erb

```diff
@@ -116,6 +149,53 @@ http {
       add_header X-XSS-Protection $xss_protection always;
       add_header Content-Type $default_content_type always;
 
+      ##
+      # Implement per-domain IP Whitelist
+      ##
+
+      access_by_lua_block {
+        local headers = ngx.req.get_headers()
+
+        -- bail fast if we don't have a whitelist, or it's not a hostname that needs filtering
+        if ip_whitelist_size == 0 then
+          return
+        end
+
+        -- where is the request coming from
+        local source_ip = ngx.var.remote_addr
+
+        -- if we have an X-Forwarded-For header then use the last one in the list instead of remote_addr
```

Comment:
Should probably link to http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html#x-forwarded-for and note this is only a safe assumption to make when behind an ELB in case this ever gets used not on AWS, or TLS termination strategy changes